onnxruntime/core/optimizer/qdq_transformer/selectors_actions/qdq_selectors.h

```diff
@@ -12,39 +12,51 @@ class Graph;
 class Node;
 
 namespace QDQ {
+
+// Struct to represent a DQ->Op->Q node group
+struct NodeGroup {
+  std::vector<NodeIndex> dq_nodes;
+  std::vector<NodeIndex> q_nodes;
+  NodeIndex target_node;
+};
+
 // Base QDQ checker. Finds and provides the DQ and Q nodes to the operator specific checkers, as the QDQ optimizations
 // always involve those nodes.
 class BaseSelector : public NodeSelector {
  public:
-  bool Select(Graph& graph, const Node& node, std::unique_ptr<NodesToOptimize>& selection) const override;
+  bool Select(const GraphViewer& graph_viewer, const Node& node,
+              std::unique_ptr<NodesToOptimizeIndexes>& selection) const override;
+
+  // This select is a QDQ Selectors only function, which takes a const GraphViewer
+  virtual bool Select(const GraphViewer& graph_viewer, const Node& node, NodeGroup& selection) const;
```

Comment:
>virtual [](http://example.com/codeflow?start=2&length=7)

does it need to be virtual and public?

Reply:
Probably don't need to be virtual, but will keep it as public, since for NNAPI we can use QDQSelector directly instead of NodeSelector